bundles/org.openhab.binding.bluetooth.bluegiga/src/main/java/org/openhab/binding/bluetooth/bluegiga/internal/BlueGigaSerialHandler.java

```diff
@@ -165,10 +76,28 @@ public void run() {
                     throw new IllegalStateException("BlueGiga handler thread failed to start");
                 }
             } catch (InterruptedException ignore) {
-                /* ignore */ }
+                /* ignore */
+            }
         }
     }
 
+    private void flush() {
+        // Send End Procedure command to end all activity and flush input buffer to start from know state
+        logger.trace("Flush input stream");
+        sendFrame(new BlueGigaEndProcedureCommand(), false);
+        try {
+            // Wait BlueGiga controller have stopped all activity
+            Thread.sleep(100);
```

Comment:
Is there a better way to do this than a Thread.sleep? Is there an easily checkable condition you can wait for?

Reply:
Unfortunately no, as we want to wait that controller have stopped sending any frames to the binding and then we can flush the buffer to start frame synchronisation.